docs/developer-guide/nodejs-api.md

```diff
@@ -109,7 +118,7 @@ You can call `verify()` like this:
 
 ```js
 const Linter = require("eslint").Linter;
-const linter = new Linter();
+const linter = new Linter({});
```

Comment:
Would you remove unnecessary change such as this line? Small diff is great thing.

Reply:
No problem!
Do you have any plan to remove the optional config object in the constructor?
Personally I don't think optional parameter is a good practice.

Reply:
No, I don't have. Because `options` is an option, people can omit it. What are the advantages to disallow it?

Reply:
Personally I prefer to limit the type as much as possible.
If something is `A | undefined` then it need to be checked whether it's `A` or `undefined` for the entire scope.
It would be a heavy burden on mind especially in javascript because there is no type check.

Reply:
For such a case, we have the default parameter syntax :)

For example, in [lib/cli-engine/file-enumerator.js#L190-L197](https://github.com/eslint/eslint/blob/master/lib/cli-engine/file-enumerator.js#L190-L197) case, users can omit options because of options, but every option isn't `undefined` in this scope explicitly.